internal/uidriver/glfw/ui_unix.go

```diff
@@ -19,10 +19,96 @@
 package glfw
 
 import (
+	"math"
+
 	"github.com/hajimehoshi/ebiten/v2/internal/driver"
 	"github.com/hajimehoshi/ebiten/v2/internal/glfw"
+	"github.com/jezek/xgb"
+	"github.com/jezek/xgb/randr"
+	"github.com/jezek/xgb/xproto"
 )
 
+type videoModeScaleCacheKey struct{ X, Y int }
+
+var videoModeScaleCache = map[videoModeScaleCacheKey]float64{}
+
+// updateMonitorsByOS must be called from the main thread.
+func updateMonitorsByOS() {
```

Comment:
`clearVideoModeScaleCache` sounds better. This is not updating.

Reply:
Done; had considered that name too but went for the generic one because the calling code doesn't know about the cache (which also only exists on one OS), but yeah, naming by what it does is of course fine too.